eigency/eigency_cpp.h

```diff
@@ -37,61 +38,62 @@ inline PyArrayObject* _ndarray_copy(const Scalar *, long rows, long cols, bool i
 }
 
-_NDAV(long double, ndarray_long_double, ndarray_long_double_C, ndarray_long_double_F)
-_NDAC(long double, ndarray_long_double, ndarray_copy_long_double_C, ndarray_copy_long_double_F)
-_NDAV(double, ndarray_double, ndarray_double_C, ndarray_double_F)
-_NDAC(double, ndarray_double, ndarray_copy_double_C, ndarray_copy_double_F)
-_NDAV(float, ndarray_float, ndarray_float_C, ndarray_float_F)
-_NDAC(float, ndarray_float, ndarray_copy_float_C, ndarray_copy_float_F)
-_NDAV(long, ndarray_long, ndarray_long_C, ndarray_long_F)
-_NDAC(long, ndarray_long, ndarray_copy_long_C, ndarray_copy_long_F)
-_NDAV(long long, ndarray_long_long, ndarray_long_long_C, ndarray_long_long_F)
-_NDAC(long long, ndarray_long_long, ndarray_copy_long_long_C, ndarray_copy_long_long_F)
-_NDAV(unsigned long, ndarray_ulong, ndarray_ulong_C, ndarray_ulong_F)
-_NDAC(unsigned long, ndarray_ulong, ndarray_copy_ulong_C, ndarray_copy_ulong_F)
-_NDAV(unsigned long long, ndarray_ulong_long, ndarray_ulong_long_C, ndarray_ulong_long_F)
-_NDAC(unsigned long long, ndarray_ulong_long, ndarray_copy_ulong_long_C, ndarray_copy_ulong_long_F)
-_NDAV(int, ndarray_int, ndarray_int_C, ndarray_int_F)
-_NDAC(int, ndarray_int, ndarray_copy_int_C, ndarray_copy_int_F)
-_NDAV(unsigned int, ndarray_uint, ndarray_uint_C, ndarray_uint_F)
-_NDAC(unsigned int, ndarray_uint, ndarray_copy_uint_C, ndarray_copy_uint_F)
-_NDAV(short, ndarray_short, ndarray_short_C, ndarray_short_F)
-_NDAC(short, ndarray_short, ndarray_copy_short_C, ndarray_copy_short_F)
-_NDAV(unsigned short, ndarray_ushort, ndarray_ushort_C, ndarray_ushort_F)
-_NDAC(unsigned short, ndarray_ushort, ndarray_copy_ushort_C, ndarray_copy_ushort_F)
-_NDAV(signed char, ndarray_schar, ndarray_schar_C, ndarray_schar_F)
-_NDAC(signed char, ndarray_schar, ndarray_copy_schar_C, ndarray_copy_schar_F)
-_NDAV(unsigned char, ndarray_uchar, ndarray_uchar_C, ndarray_uchar_F)
-_NDAC(unsigned char, ndarray_uchar, ndarray_copy_uchar_C, ndarray_copy_uchar_F)
-_NDAV(std::complex<long double>, ndarray_complex_long_double, ndarray_complex_long_double_C, ndarray_complex_long_double_F)
-_NDAC(std::complex<long double>, ndarray_complex_long_double, ndarray_copy_complex_long_double_C, ndarray_copy_complex_long_double_F)
-_NDAV(std::complex<double>, ndarray_complex_double, ndarray_complex_double_C, ndarray_complex_double_F)
-_NDAC(std::complex<double>, ndarray_complex_double, ndarray_copy_complex_double_C, ndarray_copy_complex_double_F)
-_NDAV(std::complex<float>, ndarray_complex_float, ndarray_complex_float_C, ndarray_complex_float_F)
-_NDAC(std::complex<float>, ndarray_complex_float, ndarray_copy_complex_float_C, ndarray_copy_complex_float_F)
+// This is the currently best option for this mess, tho if I can figure out a better one, I will banish this thing back to the place from whence it came
+_NDAV(signed char, __pyx_fuse_0ndarray, __pyx_fuse_0ndarray_C, __pyx_fuse_0ndarray_F)
```

Comment:
We could write this a bit more extreme in the code comment, but the thing won't compile if the order is wrong.
It would just complain, that the type is mismatched, but I totally support making this very very clear in the code comments.